tests/base/src/test/java/org/wildfly/security/ssl/TLS13AuthenticationTest.java

```diff
@@ -114,12 +116,69 @@ public void testTwoWayTLS13() throws Exception {
                 .setNeedClientAuth(true)
                 .build().create();
 
-        SecurityIdentity identity = performConnectionTest(serverContext, "protocol://test-two-way-tls13.org", "wildfly-ssl-test-config-v1_5.xml", CIPHER_SUITE);
+        SecurityIdentity identity = performConnectionTest(serverContext, "protocol://test-two-way-tls13.org", "wildfly-ssl-test-config-v1_5.xml", CIPHER_SUITE, true);
         assertNotNull(identity);
         assertEquals("Principal Name", "ladybird", identity.getPrincipal().getName());
     }
 
-    private SecurityIdentity performConnectionTest(SSLContext serverContext, String clientUri, String clientConfigFileName, String expectedCipherSuite) throws Exception {
+    @Test
+    public void testDifferentPreferredTLS13Suites() throws Exception {
+        final String REQUIRED_CIPHER_SUITE = "TLS_AES_128_GCM_SHA256";
+        final String PREFERRED_CIPHER_SUITE = "TLS_AES_256_GCM_SHA384";
+
+        SSLContext serverContext = new SSLContextBuilder()
+                .setSecurityDomain(securityDomain)
+                .setCipherSuiteSelector(CipherSuiteSelector.fromNamesString(REQUIRED_CIPHER_SUITE).add(PREFERRED_CIPHER_SUITE))
+                .setKeyManager(getKeyManager("/jks/scarab.keystore"))
+                .setTrustManager(getCATrustManager())
+                .setNeedClientAuth(true)
+                .build().create();
+
+        SecurityIdentity identity = performConnectionTest(serverContext, "protocol://test-different-preferred-tls13-suites.org", "wildfly-ssl-test-config-v1_5.xml", REQUIRED_CIPHER_SUITE, true);
+        assertNotNull(identity);
+        assertEquals("Principal Name", "ladybird", identity.getPrincipal().getName());
+    }
+
+    @Test
+    public void testServerOnlyTLS13() throws Exception {
+        final String SERVER_CIPHER_SUITE = "TLS_AES_128_GCM_SHA256";
+        final String CLIENT_CIPHER_SUITE = "TLS_RSA_WITH_AES_128_CBC_SHA256"; // TLS v1.2
```

Comment:
Since these two cipher suites are both being enabled on the server side, I think the variable names could be updated to TLS13_CIPHER_SUITE and TLS12_CIPHER_SUITE to make it more clear.